src/cwe_checker_lib/src/checkers/cwe_78/state/mod.rs

```diff
@@ -206,10 +210,24 @@ impl State {
     }
 
     /// Gets the string constant saved at the given address and saves it to the string constants field.
-    pub fn evaluate_constant(&mut self, constant: Bitvector) {
-        // TODO: check whether the constant is a valid memory address in the binary
-        // If so, get the string constant at that memory address and save it in the state
-        self.string_constants.insert(constant);
+    pub fn evaluate_constant(
+        &mut self,
+        runtime_memory_image: &RuntimeMemoryImage,
+        constant: Bitvector,
+        pointer_byte_size: ByteSize,
+    ) {
+        if runtime_memory_image.is_global_memory_address(&constant) {
+            if let Ok(parsed_address) =
+                runtime_memory_image.parse_address_if_recursive(&constant, pointer_byte_size)
+            {
+                match runtime_memory_image.read_string_until_null_terminator(&parsed_address) {
+                    Ok(format_string) => {
+                        self.string_constants.insert(format_string.to_string());
+                    }
+                    Err(e) => panic!("{}", e),
```

Comment:
I am unsure whether this should panic or not. Could this also trigger for cases, where there is no bug on the cwe_checker side, but the input binary is malformed?